docs/developer-guide/project-structure.md

```diff
@@ -1,59 +1,118 @@
-        Contains the Lucene indexing code.
-    * [strongbox-storage-metadata]/  
-        Contains a service wrapper and XML marshalling/unmarshalling for Maven metadata.
-* [strongbox-testing]/  
-    Contains various very useful base classes for testing
-    * [strongbox-testing-core]/  
-        Contains code for generation of valid Maven artifacts.
-    * [strongbox-testing-web]/  
-        Contains a dummy implementation of a Jersey application. Sometimes useful for lightweight tests.
-* [strongbox-web-core]/  
+
+    * [strongbox-storage-api]
+        Contains the the storage API.
+
+    * [strongbox-storage-core]
+        Contains the core classes for the storage API.
+        
+    * [strongbox-storage-layout-providers]
+
+        * [strongbox-storage-maven-layout]
+
+            * [strongbox-maven-metadata-api]
+                This is the implementation of the support for the `maven-metadata.xml` format.
+
+            * [strongbox-storage-maven-layout-provider]
+                This is the implementation of the [Maven 2 layout provider][Maven Layout Provider](https://strongbox.github.io/developer-guide/layout-providers/nuget-layout-provider.html). It depends on the [strongbox-maven-metadata-api](https://github.com/strongbox/strongbox/tree/master/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api).
+
+    * [strongbox-storage-npm-layout-provider]
+        This is the implementation of the [NPM layout provider](https://strongbox.github.io/developer-guide/layout-providers/npm-layout-provider.html). It depends on the [strongbox-npm-metadata](https://github.com/strongbox/strongbox-npm-metadata) project.
+
+    * [strongbox-storage-nuget-layout-provider]
+        This is the implementation of the [Nuget layout provider](https://strongbox.github.io/developer-guide/layout-providers/npm-layout-provider.html).
+        
+    * [strongbox-storage-p2-layout-provider]
+        This is an incomplete early draft implementation of the P2 OSGi layout provider. This module needs a lot more work, before the P2 layout provider could be usable.
+
+    * [strongbox-storage-raw-layout-provider]
+        This is the implementation of the [Raw layout provider](https://strongbox.github.io/developer-guide/layout-providers/raw-layout-provider.html).
+        
+* [strongbox-testing]
+    Contains various very useful base classes for testing.
+
+    * [strongbox-testing-core]
+        Contains common testing-related code.
+
+    * [strongbox-testing-storage]
+
+    * [strongbox-testing-web]
+        Contains the common `restassured` configuration for controller tests.
+
+* [strongbox-web-core]
     This is the web module which contains all the controllers.
-    
+
+* [strongbox-web-forms]
+    This is the module that contains all the web forms. You might find the [Writing Web Form Tests](https://strongbox.github.io/developer-guide/writing-web-form-tests.html) useful.
```

Comment:
This should not point to the domain. It should use a relative path to the `writing-web-form-tests.md` file. Then `mkdocs` will automatically create the right link.


```suggestion
    This is the module that contains all the web forms. You might find the [Writing Web Form Tests] useful.
    // bottom of the page
    [Writing Web Form Tests]: ./writing-web-form-tests.md
```